router/image/templates/nginx.conf

```diff
@@ -340,6 +340,157 @@ http {
             return 404;
         }
     }
+    #start k8s apps
+    {{ range $kapp := ls "/registry/services/specs/default" }}
+    {{ $upstreams := printf "/registry/services/specs/default/%s" $kapp}}
+    upstream {{ $kapp }} {
+        {{ if exists "/deis/router/affinityArg" }}
+        hash $arg_{{ getv "/deis/router/affinityArg" }} consistent;
+        {{ end }}
+        {{$appdir := printf "/registry/services/specs/default/%s" $kapp}}{{range gets $appdir}}
+        server {{$data := json .Value}}{{$data.spec.clusterIP}}:80;
```

Comment:
in some cases (like the kubernetes service that uses the port 443) this will not work.